In [1]:
math = [80, 92, 75, 60, 88]
physics = [70, 65, 90, 85, 78]
english = [95, 100, 85, 90, 87]

subjects = {
    "Math": math,
    "Physics": physics,
    "English": english
}

averages = {}

for name, scores in subjects.items():
    avg = sum(scores) / len(scores)
    averages[name] = avg

    print(
        name,
        "Average:", avg,
        "Min:", min(scores),
        "Max:", max(scores)
    )

best_subject = max(averages, key=averages.get)

print("Best subject:", best_subject)

Math Average: 79.0 Min: 60 Max: 92
Physics Average: 77.6 Min: 65 Max: 90
English Average: 91.4 Min: 85 Max: 100
Best subject: English


| Предмет | Середній бал | Мінімальний | Максимальний |
|----------|--------------|-------------|--------------|
| Math | 79.0 | 60 | 92 |
| Physics | 77.6 | 65 | 90 |
| English | 91.4 | 85 | 100 |

Найвищий середній результат: English